### PR TITLE
NetApp SDK initial version

### PR DESCRIPTION
**Pull request by leonardbf**
From new swagger spec https://github.com/Azure/azure-rest-api-specs/pull/4891
(NetApp internal ticket NFSAAS-1504)

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@leonardbf you are releasing your first SDK + your swagger is still in preview.
Please make changes as suggested to publish your SDK in preview.
Please use the generate.ps1 to generate your SDK.
If you use generate.ps1, it creates the appropriate meta data that is required for the SDK.

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/NetApp/Management.NetApp/Microsoft.Azure.Management.NetApp.csproj [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    <PropertyGroup>
+        <PackageId>Microsoft.Azure.Management.NetApp</PackageId>
+        <Description>Provides NetApp storage management capabilities for Microsoft Azure.</Description>
+        <AssemblyTitle>Microsoft Azure NetApp Management</AssemblyTitle>
+        <AssemblyName>Microsoft.Azure.Management.NetApp</AssemblyName>
+        <Version>1.0.0</Version>
+        <PackageTags>MicrosoftAzure Management;NetApp</PackageTags>
+        <PackageReleaseNotes>
+            <![CDATA[
+                * Initial version
```
[the remark this answers is not in the document]
Comment (leonardbf) [reply src/SDKs/NetApp/Management.NetApp/Microsoft.Azure.Management.NetApp.csproj]: ok, description added, including some examples. Note that a more formal description is pending anyway.

File: src/SDKs/NetApp/Management.NetApp/Generated/SdkInfo_MicrosoftNetApp.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  internal static partial class SdkInfo
+  {
+      public static IEnumerable<Tuple<string, string, string>> ApiInfo_MicrosoftNetApp
+      {
+          get
+          {
+              return new Tuple<string, string, string>[]
+              {
+                new Tuple<string, string, string>("NetApp", "Accounts", "2017-08-15"),
```
[the remark this answers is not in the document]
Comment (leonardbf) [reply src/SDKs/NetApp/Management.NetApp/Generated/SdkInfo_MicrosoftNetApp.cs]: The API is being made public but at this time is still subject to more change than will be expected later. Therefore prefer to leave as preview for now.

File: src/SDKs/NetApp/Management.NetApp/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
+[assembly: AssemblyProduct("Azure .NET SDK")]
+[assembly: AssemblyCopyright("${AuthorCopyright}")]
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+[assembly: NeutralResourcesLanguage("en")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
```
[the remark this answers is not in the document]
Comment (leonardbf) [reply src/SDKs/NetApp/Management.NetApp/Properties/AssemblyInfo.cs]: Done.

File: src/SDKs/NetApp/Management.NetApp/Microsoft.Azure.Management.NetApp.csproj [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      </PackageReference>
+    </ItemGroup>
+    <ItemGroup>
+      <PackageReference Include="Microsoft.Azure.ResourceManager" Version="1.1.0-preview" />
+    </ItemGroup>
+
+    <!-- Please do not move/edit code below this line -->
+    <Import Condition=" Exists('$([MSBuild]::GetPathOfFileAbove(AzSdk.RP.props))') " Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.RP.props'))" />
+    <!-- Please do not move/edit code above this line -->
+</Project>
```
[the remark this answers is not in the document]
Comment (leonardbf) [reply src/SDKs/NetApp/Management.NetApp/Microsoft.Azure.Management.NetApp.csproj]: Done.

File: src/SDKs/NetApp/NetApp.Tests/NetApp.Tests.csproj [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>NetApp.Tests</PackageId>
+    <Description>NetApp.Tests Class Library</Description>
+    <VersionPrefix>1.0.0</VersionPrefix>
```
[the remark this answers is not in the document]
Comment (leonardbf) [reply src/SDKs/NetApp/NetApp.Tests/NetApp.Tests.csproj]: Done.

File: src/SDKs/NetApp/NetApp.Tests/NetApp.Tests.csproj [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    <Authors>Microsoft Corporation</Authors>
+    <AssemblyName>NetApp.Tests</AssemblyName>
```
[the remark this answers is not in the document]
Comment (leonardbf) [reply src/SDKs/NetApp/NetApp.Tests/NetApp.Tests.csproj]: Done.

File: src/SDKs/NetApp/NetApp.Tests/ResourceTests/AccountTests.cs [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+                {
+                    Location = ResourceUtils.location,
+                };
+
+                try
+                {
+                    netAppMgmtClient.Accounts.CreateOrUpdate(netAppAccount, ResourceUtils.resourceGroup, ResourceUtils.accountName1);
+                    Assert.True(false);
+                }
+                catch (Exception ex)
```
[the remark this answers is not in the document]
Comment (leonardbf) [reply src/SDKs/NetApp/NetApp.Tests/ResourceTests/AccountTests.cs]: This was actually already considered but was unavailable. Leaving as is.

**Review by dsgouda — APPROVED** [review]
LGTM

File: src/SDKs/NetApp/NetApp.Tests/NetApp.Tests.csproj [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>NetApp.Tests</PackageId>
+    <Description>NetApp.Tests Class Library</Description>
+    <AssemblyName>NetApp.Tests</AssemblyName>
+    <Version>1.0.0-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/NetApp/NetApp.Tests/NetApp.Tests.csproj]: nit: Test projects must always be versioned as `1.0.0`

**Review by shahabhijeet — APPROVED** [review]

**Review by dsgouda — APPROVED** [review]
Looks good apart from a minor comment. Please address it in a later commit,

**Pull request merged**